### ISSUE TITLE
Regression on merge operator

### ISSUE THREAD

**Issue by pierrecregut** [mikefarah/yq#880]
Merge empty object with other seems to always give an empty object

Version of yq: >= 4.9.2 (4.9.1 is fine)
Operating system: linux (not tested on other)
Installed via: binary release

**Input Yaml**

data.yml:
```yaml
source:
  a: 1
  b: 2
```

**Command**
The command you ran:
```
yq e '.sink = {} * .source' data.yaml
```

**Actual behavior**

```yaml
source:
  a: 1
  b: 2
sink: {}
```

**Expected behavior**

```yaml
source:
  a: 1
  b: 2
sink:
  a: 1
  b: 2
```

**Additional context**
None

**Comment by pierrecregut** [comment]
As a side note ```yq e '{} * .source' data.yml``` gives the right result. So the title is a little misleading: the assignment is part of the problem.

**Comment by pierrecregut** [comment]
According to git bisect commit 77edbb9f5cf4d5b20fdf0db6b02f9a216d15bf5c is the culprit. It is not related to {} as I thought earlier due to my initial use case, you can put whatever you want as lhs of * operator.

**Comment by mikefarah** [comment]
Thanks for the clear issue - found the bug, will be fixed in the next release

**Comment by mikefarah** [comment]
Fixed in 4.9.7
